eth_account/signers/ledger.py

```diff
@@ -0,0 +1,362 @@
+
+        return result
+
+    def _send_to_device(self, apdu):
+        if self.device is None:
+            self.device = LedgerUsbDevice()
+
+        reply = self.device.exchange(apdu, timeout=60)
+
+        return reply
+
+    @property
+    def address(self):
+        return self.get_address(self.account_id)
+
+    def get_address(self, account_id):
+        '''
+        Query the ledger device for an ethereum address in HD wallet.
+        Offset is the number in the HD wallet tree
+        '''
+        bip32_path = self._path_to_bytes(self.path_prefix + str(account_id))
+
+        # https://github.com/LedgerHQ/blue-app-eth/blob/master/doc/ethapp.asc#get-eth-public-address
+        apdu = bytes.fromhex('e0020000')
+        apdu += bytes([len(bip32_path)])
+        apdu += bip32_path
+
+        result = self._send_to_device(apdu)
+
+        # Parse result
+        offset = 1 + result[0]
+        address = result[offset + 1: offset + 1 + result[offset]]
+        address = address.decode()  # Use decode() to convert from bytearray
+
+        return eth_utils.to_normalized_address(address)
+
+    def get_account_id(self, address, search_limit=20, search_account_id=0):
+        '''
+        Convert an address to the HD wallet tree account_id
+        Start search at an account_id. This would allow to search deeper if required.
+        Default search_limit at 20 take about 5s to reach.
+        '''
+        address = eth_utils.to_normalized_address(address)
+
+        for account_id in itertools.count(start=search_account_id):
+            if account_id > search_limit:
+                raise ValueError(f'Address {address} not found' +
+                                 f'(search_limit={search_limit}, ' +
+                                 f'search_account_id={search_account_id})')
+            if eth_utils.is_same_address(address, self.get_address(account_id)):
+                return account_id
+
+    def get_addresses(self, limit=5, page=0):
+        '''
+        List Ethereum HD wallet adrress of the ledger device
+        '''
+        return [self.get_address(account_id)
+                for account_id in range(page * limit, (page + 1) * limit)]
+
+    def signHash(self, message_hash):
+        '''
+        Not available with a Ledger
+        '''
+        raise NotImplementedError('signHash() is not available within ledger devices')
+
+    def signTransaction(self, transaction_dict):
+        '''
+        Sign a transaction, as in :meth:`~eth_account.account.Account.signTransaction`
+        but without specifying the private key.
+        '''
+        bip32_path = self._path_to_bytes(self.path_prefix + str(self.account_id))
+
+        unsigned_transaction = serializable_unsigned_transaction_from_dict(transaction_dict)
+        rlp_encoded_tx = rlp.encode(unsigned_transaction)
+
+        # https://github.com/LedgerHQ/blue-app-eth/blob/master/doc/ethapp.asc#sign-eth-transaction
+        apdu = bytes.fromhex('e0040000')
```

Comment:
'e0040000' could be saved to a constant, with a descriptive name like: `ADPU_SIGN_TX_FIRST_DATA_BLOCK`